### PR TITLE
api with promise

### PR DESCRIPTION
**Pull request by nikobenedikt**
please have a look - tried to solve it with Promise.

### CODE REVIEW

**Comment by meibegger** [comment]
Congrats! Looks OK and seems to work ... any problems?

File: src/assets/js/modules/services/api.js [hunk 1]
```diff
@@ -1,48 +1,80 @@
...
-function letsGETit(callback, URL){
-  const xhr = new XMLHttpRequest();
-  xhr.timeout = TIME_OUT;   
-  xhr.open('GET', URL);
+    //Calling make Request ->Returns Promise!
+    /**
+     * @todo 2 requests hintereinander durchführen für text und bg
+     */
+    makeRequest(scURL)
+    .then( (posts) => {
```
Comment (meibegger) [inline src/assets/js/modules/services/api.js]: Promise chaining (https://developer.mozilla.org/de/docs/Web/JavaScript/Guide/Using_promises#Chaining)
```
doSomething()
.then(result => doSomethingElse(result))
.then(newResult => doThirdThing(newResult))
.then(finalResult => {
  console.log(`Got the final result: ${finalResult}`);
})
.catch(failureCallback);
```

Comment (nikobenedikt) [reply src/assets/js/modules/services/api.js]: okay got it

File: src/assets/js/modules/services/api.js [hunk 2]
```diff
@@ -1,48 +1,80 @@
...
+      console.log("YUHU" + posts);
```
Comment (meibegger) [inline src/assets/js/modules/services/api.js]:
```
console.log('YHUHU', posts);
```
or better
```
console.log('YHUHU', result);
```

Comment (meibegger) [inline src/assets/js/modules/services/api.js]: don't mix `'` and `"` ... always use `'`

File: src/assets/js/modules/services/api.js [hunk 3]
```diff
@@ -1,48 +1,80 @@
...
+      callback();
```
Comment (meibegger) [inline src/assets/js/modules/services/api.js]: pass some argument to the callback function and then handle the result there

```
callback(true)

function callback(success) {
  if (success) {
    alert('SUCCESS');
  } else {
    alert('FAILED');
  }
}
```

**Pull request closed without merging**